mantis-runtime/src/main/java/io/mantisrx/runtime/executor/StageExecutors.java

```diff
@@ -313,7 +324,9 @@ private static int resolveStageConcurrency(int givenStageConcurrency) {
         if (givenStageConcurrency == StageConfig.DEFAULT_STAGE_CONCURRENCY) {
             String jobParamPrefix = "JOB_PARAM_";
             String stageConcurrencyParam = jobParamPrefix + STAGE_CONCURRENCY;
-            String concurrency = System.getenv(stageConcurrencyParam);
+            // Need to be compatible for both Mesos and TaskExecutor.
+            String concurrency = System.getProperty(stageConcurrencyParam, System.getenv(stageConcurrencyParam));
```

Comment:
I think we should avoid both System::getEnv as well as System::getProperty.

Reply:
this might be tricky in some places where they don't have any injection/context to use. Any thought on best practice here?

Reply:
As discussed offline, let's make use of the Context class to get this value.

Reply:
fixed (read via context now).